### PR TITLE
Remove repos/jikesrvm submodule

### PR DESCRIPTION
**Pull request by qinsoon**
This PR removes the git submodule for JikesRVM. This PR closes https://github.com/mmtk/mmtk-jikesrvm/issues/89.

Changes:
* Removed JikesRVM git submodule. The metadata about the JikesRVM version is now stored in `mmtk/Cargo.toml`.
* Added `common.sh` to `.github/scripts`, which is included by each CI shell script.
* Added `ci-checkout.sh` to `.github/scripts` that checks out the correct OpenJDK version for CI.
* Updated `README`.

### CODE REVIEW

File: .github/scripts/ci-checkout.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# ci-checkout.sh
+set -ex
+
+. $(dirname "$0")/common.sh
+
+JIKESRVM_URL=`sed -n 's/^jikesrvm_repo.=."\(.*\)"$/\1/p' $BINDING_PATH/mmtk/Cargo.toml`
+JIKESRVM_VERSION=`sed -n 's/^jikesrvm_version.=."\(.*\)"$/\1/p' $BINDING_PATH/mmtk/Cargo.toml`
```
Comment (wks) [inline .github/scripts/ci-checkout.sh:7]: Should parse it as TOML. See the `mmtk-openjdk` PR.

File: README.md [hunk 2]
```diff
@@ -41,22 +41,59 @@ $ ssh-add
...
+To work on JikesRVM binding, we expect you have a directory structure like below. This section gives instructions on how to check out
+those repositories with the correct version.
+
+```
+Your working directory/
+├─ mmtk-jikesrvm/
+│  ├─ jikesrvm/
+│  ├─ mmtk/
+├─ jikesrvm/
+├─ mmtk-core/ (optional)
```
Comment (wks) [inline README.md:53]: May use the elbow-shaped unicode character like the `mmtk-openjdk` binding.

**Review by wks — COMMENTED** [review]
There are similar problems as mmtk-openjdk

**Review by wks — APPROVED** [review]
LGTM

**Comment by qinsoon** [comment]
I would suggest merging this pull request (allow merging this PR without all tests passed), as 1. the problem is not introduced by this pull request, and 2. without this pull request, we cannot run proper CI tests for the binding. I will do a pull request to fix the problem. What do you think? @wks

**Comment by wks** [comment]
@qinsoon I think it is OK to merge it now.  It temporarily breaks some tests, but as long as it doesn't take long to provide a proper fix, it should be OK.

**Pull request merged**